pkg/domain/infra/abi/play.go

```diff
@@ -985,6 +993,32 @@ func (ic *ContainerEngine) playKubePVC(ctx context.Context, pvcYAML *v1.Persiste
 	return &report, nil
 }
 
+func (ic *ContainerEngine) mergePublishPorts(p *entities.PodCreateOptions, publishPortsOption []nettypes.PortMapping) {
+	// Make a map out of the slice for fast lookup
+	publishPortsOptionMap := make(map[uint16]nettypes.PortMapping, len(publishPortsOption))
+	for _, publishPortOption := range publishPortsOption {
+		publishPortsOptionMap[publishPortOption.ContainerPort] = publishPortOption
+	}
+
+	updatedPublishPorts := make([]nettypes.PortMapping, 0, len(p.Net.PublishPorts))
+	for _, publishPortSpec := range p.Net.PublishPorts {
+		// If exists in both, command line takes precedence
+		if publishPortOption, ok := publishPortsOptionMap[publishPortSpec.ContainerPort]; ok {
+			updatedPublishPorts = append(updatedPublishPorts, publishPortOption)
+			delete(publishPortsOptionMap, publishPortSpec.ContainerPort)
+		} else {
+			updatedPublishPorts = append(updatedPublishPorts, publishPortSpec)
+		}
+	}
+
+	// Add ports from Options that do not exist in spec
+	for _, publishPortOption := range publishPortsOptionMap {
+		updatedPublishPorts = append(updatedPublishPorts, publishPortOption)
+	}
+
+	p.Net.PublishPorts = updatedPublishPorts
+}
```

Comment:
I've changed the code so that it starts with the list of published ports specified in the command line and adds to the list ports from the YAML file that are not listed in the command line. The matching is still done according to the `ContainerPort`.